go-controller/pkg/ovn/gateway_init.go

```diff
@@ -287,6 +259,19 @@ func gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet, hostSubnets []*n
 		}
 	}
 
+	// We need to add a route to the Gateway router's IP, on the
+	// cluster router, to ensure that the return traffic goes back
+	// to the same gateway router
+	for _, gwLRPIP := range gwLRPIPs {
+		stdout, stderr, err = util.RunOVNNbctl("--may-exist", "lr-route-add",
```

Comment:
@trozet reply packets from the pod to external requests should be send back to the GR from where the request packets ingress the OVN logical topology.  If you don't have this /32 routes, then the reply packet from a Pod will be sent to GR on the pod's node. To achieve symmetric routing this is required. 

We didn't need it with multiple join switches because we use /29 subnet for each of the join switch. Due to longest prefix match rule, the /29 routes in ovn_cluster_router will get matched before the source-ip based routes.

Reply:
The incoming packet would be SNAT'ed to the GR IP address. When the reply packet comes back, it is already destined for to the right GR. ovn_cluster_router is directly attached to the subnet where that GR resides. It doesn't need a static route, it already knows where to send the packet because it has a directly connected route to that subnet. It just arps and sends the packet directly to the GR. If this really is required, it seems like a bug in OVN to me. Ping me on slack and we can discuss further.

Reply:
After talking it through with Girish on Friday he clarified why we need this. I went ahead and filed: https://bugzilla.redhat.com/show_bug.cgi?id=1891516

@cathy-zhou can you please add a comment here to remove this route after 1891516 is fixed?

Reply:
done. thanks @trozet